paasta_tools/kubernetes_tools.py

```diff
@@ -542,7 +542,6 @@ def get_autoscaling_metric_spec(
                 )
             )
         elif metrics_provider in ("http", "uwsgi"):
-            annotations = {"signalfx.com.custom.metrics": ""}
```

Comment:
You may want to go ahead remove the `else` block starting at line 582, since I think that functionality will be broken by this removal. (Or add this annotation within that block)

Reply:
hmm, I wonder if we should also remove the 
```
if (
    autoscaling_params.get("forecast_policy") == "moving_average"
    or "offset" in autoscaling_params
    or load_system_paasta_config().get_hpa_always_uses_external_for_signalfx()
)
``` 
check since that'll always be true anyway - but I guess we can clean that up when I fully remove http or when we cleanup once we're done here?